src/inmanta/agent/agent.py

```diff
@@ -1136,14 +1141,28 @@ def periodic_schedule(
                     interval=float(interval), initial_delay=float(splay_value)
                 )
                 self._enable_time_trigger(action, interval_schedule)
+                return True
 
             if isinstance(interval, str):
                 self.logger.info("Scheduling periodic %s with cron expression '%s'", kind, interval)
                 cron_schedule = CronSchedule(cron=interval)
                 self._enable_time_trigger(action, cron_schedule)
+                return True
+            return False
 
         now = datetime.datetime.now().astimezone()
-        periodic_schedule("deploy", deploy_action, self._deploy_interval, self._deploy_splay_value, now)
+        did_schedule = periodic_schedule("deploy", deploy_action, self._deploy_interval, self._deploy_splay_value, now)
+        if not did_schedule and self.ensure_deploy_on_start:
```

Comment:
Why only if there's no schedule? If the agent were already up it would've gotten an explicit deploy instruction, wouldn't it? If the auto deploy is a cron expression it might not deploy for hours, or even much longer.

Reply:
I checked if this is safe and it seems so, so I updated it